storage/cache/subtree_cache.go

```diff
@@ -196,7 +201,25 @@ func (s *SubtreeCache) preload(ids []storage.NodeID, getSubtrees GetSubtreesFunc
 	for _, t := range subtrees {
 		s.populate(t)
 		s.subtrees[string(t.Prefix)] = t
+		delete(want, string(t.Prefix))
 	}
+
+	// We might not have got all the subtrees we requested, if they don't already exist.
+	// Create empty subtrees for anything left over. As an additional sanity check we refuse
+	// to overwrite anything already in the cache as we determined above that these subtrees
+	// should not exist in the subtree cache map.
+	for _, id := range want {
+		prefixLen := id.PrefixLenBits / depthQuantum
+		px := id.Path[:prefixLen]
+		t := s.newEmptySubtree(*id, px)
```

Comment:
Move this creation down to L217 where it's used (handily avoids creating it if we don't need to)?